techmatters/terraso-mobile-client
# feat: delete account screen

## Description

Implement the user account deletion request screen, including integration with backend system to handle the deletion-request user preference. Also do some refactoring of related code for better organization and clarity.

### Checklist

### Related Issues
#1263

### Verification steps

Open settings screen and press 'delete account' button. Verify that you can submit a request when you enter your email address, and cannot submit a request until then. Verify that the screen changes to the pending content on submission, and that when you return to the screen it continues to display the pending message.
